### PR TITLE
fix: handle Google Calendar UNTIL without time respecting event timezone

### LINKED ISSUE #435: Throws UNTIL rule part MUST have the same value type as DTSTART
Up until **node-ical** v0.22.1 parsing went fine.

With **node-ical** v0.23.1 it has started throwing `UNTIL rule part MUST have the same value type as DTSTART`.

I know this is comming from rrule-temporal, but it seems that the problem is how node-ical calls upon rrule-temporal.

With the ics event shown below, node-ical throws `UNTIL rule part MUST have the same value type as DTSTART` via rrule-temporal.

However, when using the [rrule-temporal Playground](https://ggaabe.github.io/rrule-temporal/) and inserting the **DTSTART** and the **RRULE** in the `Raw` section, it doesn't complain:
```text
DTSTART;VALUE=DATE:20231027
RRULE:FREQ=WEEKLY;UNTIL=20240523
```

Ics event:
```text
BEGIN:VEVENT
DTSTART;VALUE=DATE:20231027
DTEND;VALUE=DATE:20231028
RRULE:FREQ=WEEKLY;UNTIL=20240523
EXDATE;VALUE=DATE:20240517
DTSTAMP:20260131T113350Z
UID:202B21E4-0844-44D2-ABA1-3528A7ABC85D
CREATED:20240508T202621Z
LAST-MODIFIED:20240523T065534Z
SEQUENCE:0
STATUS:CONFIRMED
SUMMARY:Test av gammal heldags Gmail recurr
TRANSP:OPAQUE
X-APPLE-CREATOR-IDENTITY:com.apple.mobilecal
X-APPLE-CREATOR-TEAM-IDENTITY:0000000000
END:VEVENT
```

The error thrown:
```text
/Users/runely/sources/private/calendar-homey-test/node_modules/rrule-temporal/dist/index.cjs:197
          throw new Error("UNTIL rule part MUST have the same value type as DTSTART");
          ^

Error: UNTIL rule part MUST have the same value type as DTSTART
    at parseRRuleString (/Users/runely/sources/private/calendar-homey-test/node_modules/rrule-temporal/dist/index.cjs:197:17)
    at new _RRuleTemporal (/Users/runely/sources/private/calendar-homey-test/node_modules/rrule-temporal/dist/index.cjs:258:22)
    at Object.END (/Users/runely/sources/private/calendar-homey-test/node_modules/node-ical/ical.js:891:33)
    at Object.handleObject (/Users/runely/sources/private/calendar-homey-test/node_modules/node-ical/ical.js:944:39)
    at Object.parseLines (/Users/runely/sources/private/calendar-homey-test/node_modules/node-ical/ical.js:996:18)
    at Immediate.<anonymous> (/Users/runely/sources/private/calendar-homey-test/node_modules/node-ical/ical.js:1011:16)
    at process.processImmediate (node:internal/timers:504:21)
```

When adding a try/catch around ical.js:891-896:
```JavaScript
try {
  const rruleTemporal = new RRuleTemporal({
    rruleString: rruleOnly,
    dtstart: dtstartTemporal,
  });

  curr.rrule = new RRuleCompatWrapper(rruleTemporal);
} catch (_err) {
  console.error(`[node-ical] Failed to parse RRULE: '${rruleOnly}' with dtstartTemporal: '${dtstartTemporal.toString()}' with original DTSTART: '${curr.start}' on UID: '${curr.uid}'`);
  throw _err;
}
```

it console errors out this:
```text
[node-ical] Failed to parse RRULE: 'FREQ=WEEKLY;UNTIL=20240523' with dtstartTemporal: '2023-10-27T00:00:00+00:00[UTC]' with original DTSTART: 'Fri Oct 27 2023 00:00:00 GMT+0000 (Coordinated Universal Time)' on UID: '202B21E4-0844-44D2-ABA1-3528A7ABC85D'
```

### PR DESCRIPTION
**Pull request by KristjanESPERANTO**
## Background

In #422, we switched from the unmaintained `rrule` library to `rrule-temporal`. This remains a good choice: active maintenance, strict RFC5545 validation catches bugs, and Temporal API enables accurate timezone handling. However, strict validation exposes non-conformant calendar data in the wild that `rrule` silently accepted. Non-conformant providers should fix their exports, but as a parser `node-ical` has to handle real-world data pragmatically by normalizing before validation.

## Problem

Google Calendar produces invalid RRULEs violating RFC5545 § 3.3.10 (UNTIL must match DTSTART value type):

```ics
DTSTART;TZID=Europe/Oslo:20211216T180000
RRULE:FREQ=WEEKLY;UNTIL=20211216          ← DATE-only, should be DATE-TIME
```

PR #433 fixed the opposite case (DATE DTSTART with time in UNTIL), but missed this direction.

## Solution

Normalize UNTIL before passing to rrule-temporal: when DTSTART is DATE-TIME but UNTIL is DATE-only, interpret UNTIL as 23:59:59 in the event's timezone, then convert to UTC.

**Critical for eastern timezones:** Naive `T235959Z` (UTC) fails for Pacific/Auckland (UTC+13) where Dec 16 23:59:59 UTC = Dec 17 12:59 Auckland, incorrectly including Dec 17 events.

## Tests

Added 5 comprehensive tests including Pacific/Auckland edge case.

## Fixes

Fixes #435

Related: ggaabe/rrule-temporal#104

## Summary by CodeRabbit

* **Bug Fixes**
  * Corrected handling of recurrence UNTIL values that lack a time: treat UNTIL as end-of-day in the event’s local timezone and convert to UTC when possible, preventing off-by-one-day recurrences.

* **Tests**
  * Added regression tests covering UNTIL-without-time across multiple timezones and recurrence scenarios to ensure correct occurrence boundaries.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: ical.js [hunk 1]
```diff
@@ -808,6 +808,34 @@ module.exports = {
...
+                      rruleOnly = rruleOnly.replace(/UNTIL=(\d{8})(?!T)/, `UNTIL=${untilUtc}`);
+                      converted = true;
+                    }
+                  } catch {/* Fall through to UTC fallback */}
+                }
+
+                if (!converted) {
+                  // No timezone info available - assume UNTIL date means end-of-day UTC
+                  rruleOnly = rruleOnly.replace(/UNTIL=(\d{8})(?!T)/, 'UNTIL=$1T235959Z');
+                }
```
[the remark this answers is not in the document]
Comment (KristjanESPERANTO) [reply ical.js:838]: Done.

**Comment by KristjanESPERANTO** [comment]
@jens-maus Seems coderabbit is satisfied :slightly_smiling_face: 

Sorry that the migration to `rrule-temporal` is causing such a trail of follow-up PRs.

**Comment by jens-maus** [comment]
No problem. Something like this happens and I already expected something like that.
